index.bs

```diff
@@ -461,6 +519,51 @@ steps:
 
 </div>
 
+### The {{FileSystemWriter/flush()}} method ### {#api-filesystemwriter-flush}
+
+<div class="note domintro">
+  : await |writer| . {{FileSystemWriter/flush()}}

```

Comment:
How about discussing flush() in a separate PR?

`flush()` for atomic files isn't intuitive to me. I'd expect that atomic=true implies that native apps on the system can only observe the file in the states marked by `flush()` and `close()` calls.  A browser would implement atomic files by writing to a temporary file, and atomically renaming to the destination file on `close()`. `flush()` would have to be implemented using the steps for `close()`, followed by copying the file into a temporary that receives future writes.

If the above is true, I'd rather that apps explicitly close the writer and create a new one.

`flush()` for in-place writers makes more sense, but feels like an optimization. Without flush, the app would have to close the writer and create a new writer, whereas `flush()` saves a bit of file descriptor churn.

In the in-place case, `flush()` adds a bunch of complexity.

1) Without `flush()`, we can say that the changes made to a file are guaranteed to be visible from native apps when the writer is closed (valid for all modes and options). This lets implementation grab an exclusive lock on the file, and release the lock after doing security checks. 
2) The name `flush()` is close enough to `FlushFileBuffers` that developers may assume it acts like `fsync()` and prevents data loss on power failure. While we'll need to discuss guarantees in this spec anyway, I'd prefer that we stay away from all that `flush()` implies until we have a use case where creating a new writer doesn't cut it.

Reply:
Good point, I'll remove flush for now, and I'm mostly convinced we probably don't need it anyway.